### PR TITLE
Add introduction slides of `epi_slide`

### PR DESCRIPTION
**Pull request by XuedaShen**
Added slides introducing `epi_slide()` and `epix_slide()`. 

Introduced `epi_slide()` with a trailing 7 day average as an example, discussed how to do it via tidy evaluation and via custom function. Omitted sliding by formula because a). we didn't use it anywhere (I believe?). b). the syntax for that is a bit terse to explain.

`epix_slide()` is introduced very briefly during the introduction of `epi_archive()`. 

EDIT: commit `76872a1` fixes visualization of revision pattern of the `epi_archive` example. More details in commit message.

### CODE REVIEW

File: slides/day1-afternoon.qmd [hunk 1]
```diff
@@ -696,6 +696,7 @@ attr(edf, "metadata")
 1. Correlating signals across location or time 
 1. Computing growth rates
 1. Detecting and removing outliers
+1. Applying computation with rolling window
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:699]:
```suggestion
1. Calculating summaries with rolling windows
```

File: slides/day1-afternoon.qmd [hunk 2]
```diff
@@ -864,6 +865,136 @@ edfo |>
 ```
 
 
+## Features -- sliding computation on a `epi_df`
+
+* Many aggregations on panel data applies some function to a rolling window of data. 
+
+* By sliding the window, we could apply function to different parts of the snapshot. 
+
+* For example, trailing average is often used to smooth out noises in individual data points.
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:874]:
```suggestion
* It is often useful to compute rolling summaries of signals. 

* These depend on the reference time, and are computed separately over geographies (and other groups). 

* For example, a trailing average can smooth out daily variation.
```

File: slides/day1-afternoon.qmd [hunk 3]
```diff
@@ -864,6 +865,136 @@ edfo |>
...
+  ...,
+  .window_size = NULL,
+  .align = c("right", "center", "left"),
+  .ref_time_values = NULL,
+  .new_col_name = NULL,
+  .all_rows = FALSE
+)
+```
+
+As an example, we show how we can use `epi_slide` to compute a trailing 7-day average.
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:894]:
```suggestion
For example, we can use `epi_slide()` to compute a trailing 7-day average. 
```

File: slides/day1-afternoon.qmd [hunk 4]
```diff
@@ -864,6 +865,136 @@ edfo |>
...
+
+```{r epi-slide-tidy-eval-display-result}
+#| echo: false
+
+cases_7dav |>
+  arrange(geo_value, time_value) |>
+  group_by(geo_value) |>
+  slice_min(time_value, n = 2) |>
+  as_tibble() |>
+  print()
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:934]:
```suggestion
```

File: slides/day1-afternoon.qmd [hunk 5]
```diff
@@ -864,6 +865,136 @@ edfo |>
...
+}
+```
+
+* `x`: the data frame with all the columns with original object [except]{.primary} groupping vars. 
+* `g`: the one-row tibble with values of gropping vars of the given group. 
+* `t`: the `.ref_time_value` of the current window. 
+* `...`: additional arguments. 
+
+
+## Features -- sliding computation on a `epi_df`
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:961]:
```suggestion
## Features -- sliding a computation on an `epi_df`
```

File: slides/day1-afternoon.qmd [hunk 6]
```diff
@@ -864,6 +865,136 @@ edfo |>
...
+#| echo: true
+#| code-line-numbers: "|9-17"
+
+custom_function <- function(x, g, t, ...) {
+
+  result = data.frame(cases_7dav = mean(x$raw_cases, na.rm = TRUE))
+
+  return(result)
+
+}
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:973]:
```suggestion
mean_by_hand <- function(x, g, t, ...) {
  data.frame(cases_7dav = mean(x$raw_cases, na.rm = TRUE))
}
```

File: slides/day1-afternoon.qmd [hunk 7]
```diff
@@ -864,6 +865,136 @@ edfo |>
...
+
+cases_mean_custom_f = epi_slide(
+    .x = cases_edf,
+    .f = custom_function,
+    .window_size = 7,
+    .align = "right",
+    .ref_time_values = NULL,
+    .new_col_name = NULL,
+    .all_rows = FALSE
+)
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:983]:
```suggestion
cases_mean_custom_f = epi_slide(
    .x = cases_edf,
    .f = mean_by_hand,
    .window_size = 7,
    .align = "right",
)
```

File: slides/day1-afternoon.qmd [hunk 8]
```diff
@@ -864,6 +865,136 @@ edfo |>
...
+
+```
+
+
+```{r epi-slide-custom-function-display-result}
+cases_mean_custom_f |>
+  arrange(geo_value, time_value) |>
+  slice_min(time_value, n = 2) |>
+  as_tibble() |>
+  print()
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:993]:
```suggestion
  slice_min(time_value, n = 2)
```

File: slides/day1-afternoon.qmd [hunk 9]
```diff
@@ -875,20 +1006,94 @@ edfo |>
...
+#| eval: false
+
+epix_slide(
+  .x,
+  .f,
+  ...,
+  .before = Inf,
+  .versions = NULL,
+  .new_col_name = NULL,
+  .all_versions = FALSE
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:1037]:
```suggestion
```

File: slides/day1-afternoon.qmd [hunk 10]
```diff
@@ -875,20 +1006,94 @@ edfo |>
...
+)
+
+```
+
+This functionality is very helpful in version aware forecasting. We will return with a concrete example.
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:1042]:
```suggestion
This functionality is critical for version aware forecasting, as we will see.
```

File: slides/day1-afternoon.qmd [hunk 11]
```diff
@@ -875,20 +1006,94 @@ edfo |>
...
+  .all_versions = FALSE
+)
+
+```
+
+This functionality is very helpful in version aware forecasting. We will return with a concrete example. 
   
-## Revision patterns
+
+## Features -- revision analysis helper function
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:1045]:
```suggestion
## Features -- summarize revision behavior
```

File: slides/day1-afternoon.qmd [hunk 12]
```diff
@@ -875,20 +1006,94 @@ edfo |>
...
+  quick_revision = as.difftime(3, units = "days"),
+  few_revisions = 3,
+  abs_spread_threshold = NULL,
+  rel_spread_threshold = 0.1,
+  compactify_tol = .Machine$double.eps^0.5,
+  should_compactify = TRUE
+)
+```
+
+## Features -- revision analysis helper function
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:1069]:
```suggestion
## Features -- summarize revision behavior
```

File: slides/day1-afternoon.qmd [hunk 13]
```diff
@@ -875,20 +1006,94 @@ edfo |>
...
+
+```{r epiprocess-revision-summary-results}
+#| echo: true
+
+head(revision_data)
+```
+
+
+
+## Revision pattern visualization
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:1079]:
```suggestion
## Visualize revision patterns
```

File: slides/day1-afternoon.qmd [hunk 14]
```diff
@@ -875,20 +1006,94 @@ edfo |>
...
+
+versions <- seq(as.Date("2020-07-01"), as.Date("2021-11-30"), by = "1 month")
+max_version <- max(versions)
+
+snapshots <- map(versions, function(v) {
+  res = epix_as_of(archive_cases_dv_subset_all_states, v) |>
+    mutate(version = v)} |>
+  filter(geo_value %in% c("ca", "ut"))) |>
+  bind_rows() |>
+  mutate(latest = version == max_version)
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:1093]: Check that this change works. It's hard to see what function is being mapped in the current.

```suggestion
snapshots <- map(versions, \(v) {
  epix_as_of(archive_cases_dv_subset_all_states, v) |>
    mutate(version = v) |>
    filter(geo_value %in% c("ca", "ut"))
  }) |>
  bind_rows() |>
  mutate(latest = version == max_version) 
```

File: slides/day1-afternoon.qmd [hunk 15]
```diff
@@ -1805,7 +2014,7 @@ The main steps are...
 
 3. [**Model & Prediction**]{.primary}: Fit a linear model to predict cases based on %CLI, trained on a 30-day rolling window.
 
-4. [**Nowcast Execution**]{.primary}: Use `epix_slide` to nowcast the cases dynamically. 
+4. [**Nowcast Execution**]{.primary}: Use `epix_slide` to nowcast the cases dynamically.
```
Comment (dajmcdon) [inline slides/day1-afternoon.qmd:2017]:
```suggestion
4. [**Nowcast Execution**]{.primary}: Use `epix_slide()` to nowcast the cases dynamically.
```

**Review by dajmcdon — APPROVED** [review]

**Comment by XuedaShen** [comment]
OK I think I have actually untracked all files other than `day1-afternoon.qmd`? Hopefully I have done this correctly and merge would be easier

**Comment by XuedaShen** [comment]
I am going to close this PR and start new one with qmd file only  (since this one is too gnarly to deal with and only modification is textual).

**Pull request closed without merging**